Issue #4 - rewrite import - first batch of changes

simon-20 opened the pull request:
Some initial changes to begin the import rewriting.

Some docstrings added; function renaming for easier reading; line indentation in line with PEP 8; few other simplifications

simon-20 committed:
Docstrings, renaming, some simplification

This commit includes:
- changes to indentation inline with PEP 8
- some docstrings have been added
- some of the import code has been simplified / streamlined a little

It also adds a command to create the database, because the upgrade command doesn't work after a drop command.

markbrough commented on iatidatacube/import_data.py:
```diff
@@ -79,29 +81,30 @@ def delete_dataset(csv_file, inserted_ids):
-        ).order_by(IATILine.id)
+            IATILine.recipient_country_or_region == dataset_country,
+            IATILine.transaction_type == 'budget'
+        ).order_by(IATILine.id)  # sjk: now that 'id' is a hash, its unclear whether this ordering does anything useful
```
Oh interesting yes -- I originally did this to speed up searching in the list of identifiers. But as the list of identifiers is now a set, which is unordered, I don't think this `.order_by()` now does anything?

markbrough commented on iatidatacube/import_data.py:
```diff
@@ -166,24 +169,55 @@ def row_from_csv(row, codelists, reporting_organisation,
+    :type start_at_filename: str
+    :param end_at_filename: The filename in output/csv to end processing at. If empty, run until the last file.
+    :type end_at_filename: str
+    :param force_update: If true, activities already in the DB will be deleted and re-imported
```
If true, activities already in the DB will be updated, regardless of whether the `hash` column on the activity row has changed or not. This could be useful, for example, if you have changed the import logic for activities.

simon-20 replied:
Added a note to that effect, as that's a useful point

markbrough commented on iatidatacube/import_data.py:
```diff
@@ -270,8 +305,7 @@ def add_or_update_activity(activity_row):
     if ro_row not in reporting_orgs:
-        ro_row_ref = get_groups_or_none(re.match(r"(.*) \[(.*)\]",
-                ro_row), 1)
+        ro_row_ref = get_groups_or_none(re.match(r"(.*) \[(.*)\]", ro_row), 1)
```
It would probably be useful if `iati-flattener` would add a column `reporting_org_ref` which would e.g. just contain `XM-DAC-46002` rather than `African Development Bank [XM-DAC-46002]` - I would imagine this line adds some unnecessary overhead?

simon-20 replied:
Yeah, could do. I'll add an issue.

simon-20 committed:
Refactoring to improve readability

Renamed functions to be more expressive; added some docstrings to `import_data.py` and others; brought some indentation inline with PEP 8 and other recommendations; added info for developers to `README.md`.

simon-20 committed:
Merge branch 'iati-data-access:main' into 4-rewrite-import

simon-20 committed:
Added note from @markb to docstring

Added note from @markb to docstring

markbrough reviewed (approved):
Looks great -- much clearer, thank you!
